tests/conftest.py

```diff
@@ -412,68 +414,188 @@ def chdir_tmp_path(request, tmp_path):
 
-        .. note:: the `output_lines` attribute is added to return value containing list of stripped output lines.
-
-        :param options: the list of command line options to pass to the command invocation.
+        :param command: The base command to invoke.
+        :param parameters: The command line parameters to pass to the invocation.
         :param user_input: string with data to be provided at the prompt. Can include newline characters to simulate
             responses to multiple prompts.
-        :param raises: whether the command is expected to raise an exception.
-        :param catch_exceptions: if True and ``raise is False``, will assert that the exception is ``None`` and the exit
-            code of the result of the invoked command equals zero.
-        :param kwargs: keyword arguments that will be psased to the command invocation.
-        :return: test result.
+        :param raises: Boolean, if ``True``, the command should raise an exception.
+        :param use_subprocess: Boolean, if ``True``, runs the command in a subprocess, otherwise it is run in the same
+            interpreter using :class:`click.testing.CliRunner`. The advantage of running in a subprocess is that it
+            simulates exactly what a user would invoke through the CLI. The test runner provided by ``click`` invokes
+            commands in a way that is not always a 100% analogous to an actual CLI call and so tests may not cover the
+            exact behavior. However, if a test monkeypatches the behavior of code that is called by the command being
+            tested, then a subprocess cannot be used, since the monkeypatch only applies to the current interpreter. In
+            these cases it is necessary to set ``use_subprocesses = False``.
+        :param suppress_warnings: Boolean, if ``True``, the ``PYTHONWARNINGS`` environment variable is set to
+            ``ignore::Warning``. This is important when testing a command using ``use_subprocess = True`` and the check
+            on the output is very strict. By running in a sub process, any warnings that are emitted by the code will be
+            shown since they have not already been hit as would be the case when running the test through the test
+            runner in this interpreter.
+        :returns: Instance of ``CliResult``.
+        :raises AssertionError: If ``raises == True`` and the command didn't except, or if ``raises == True`` and the
+            the command did except.
         """
-        import traceback
+        # Cast all elements in ``parameters`` to strings as that is required by ``subprocess.run``.
+        parameters = [str(param) for param in parameters or []]
 
-        from aiida.cmdline.commands.cmd_verdi import VerdiCommandGroup
-        from aiida.common import AttributeDict
+        if use_subprocess:
+            command_path = cli_command_map[command]
+            args = command_path[:1] + ['-p', aiida_profile.name] + command_path[1:] + parameters
```

Comment:
might make sense to move this into the `run_cli_command_subprocess` function (and add the aiida profile name as a function argument), so it becomes more parallel to `run_cli_command_runner`